File: Data Structures & Algorithms/three-integer-sum/submission-0.java
class Solution {
    public List<List<Integer>> threeSum(int[] nums) {
        Arrays.sort(nums);
        List<List<Integer>> ans = new ArrayList<>();
        int len = nums.length;
        for(int i=0;i<len;i++){
            if(i > 0 && nums[i] == nums[i - 1]){
                continue;
            }
            int l = i + 1,r = len - 1;
            while(l<r){
                int threeSum = nums[i] + nums[l] + nums[r];
                if(threeSum > 0){
                    r--;
                }
                else if(threeSum < 0){
                    l++;
                }
                else{
                    ans.add(Arrays.asList(nums[i],nums[l],nums[r]));
                    l++;
                    while(nums[l] == nums[l - 1] && l < r){
                        l++;
                    }
                }
            }
            // System.out.println(nums[i]);
        }
        
        return ans;
    }
}

File: Data Structures & Algorithms/single-number-iii/submission-0.py
class Solution:
    def singleNumber(self, nums: List[int]) -> List[int]:
        x = 0
        for i in nums: x ^= i
        rm = (x & (x - 1)) ^ x
        b1,b2 =0,0
        for i in nums:
            if i & rm:
                b1 ^= i
            else:
                b2 ^= i
        return [b1,b2]